tests/test_freshdesk_start_date.py

```diff
@@ -90,19 +89,31 @@ def test_run(self):
         if len(second_sync_empty) > 0:
             print("Missing stream(s): {} in sync 2. Updating expectations"\
                   .format(second_sync_empty))
-            self.second_sync_empty = second_sync_empty
+        self.second_sync_empty = second_sync_empty
         second_sync_bonus = synced_records_2.keys() - self.test_streams
         if len(second_sync_bonus) > 0:
             print("Found stream(s): {} in second sync. Add to test_streams?".format(second_sync_bonus))
 
         for stream in test_streams:
             with self.subTest(stream=stream):
 
-                if stream in self.second_sync_empty:
-                    print("No sync 2 data to compare for stream: {}, start_date obeyed".format(stream))
+                if stream in self.first_sync_empty:
+                    self.assertTrue(False, msg="Stream: {} missing from sync 1".format(stream))
 
                     continue
 
+                if stream in self.second_sync_empty:
+                    if stream == 'roles':
+                        self.assertTrue(True, msg="Expected 0 records for stream {}".format(stream))
+                        print("No sync 2 data to compare for stream: {}, start_date obeyed".format(stream))
```

Comment:
I think it would better to assert that records were replicated in a clearer assertion statement like assertGreater between the actual record count or 0 or something like that. I think it's the same result functionally though so not a big deal. This just at first glance was difficult for me to follow.

Reply:
Noted.